File: frontend-react/src/lib/passwordPolicy.ts
import { z } from 'zod'

export const PASSWORD_POLICY_HINT =
  'Minimo 8 caracteres, al menos una mayuscula y un numero.'

export function buildPasswordSchema(fieldLabel: string) {
  return z
    .string()
    .min(8, `${fieldLabel} debe tener al menos 8 caracteres.`)
    .max(72, `${fieldLabel} no puede superar 72 caracteres.`)
    .regex(/[A-Z]/, `${fieldLabel} debe incluir al menos una mayuscula.`)
    .regex(/\d/, `${fieldLabel} debe incluir al menos un numero.`)
}

File: backend-java/src/main/java/com/asistentewhatsapp/security/application/PasswordPolicy.java
package com.asistentewhatsapp.security.application;

import org.springframework.stereotype.Component;

@Component
public class PasswordPolicy {

	public void validate(String password) {
		if (password == null || password.length() < 8 || password.length() > 72) {
			throw new IllegalArgumentException("La contrasena debe tener entre 8 y 72 caracteres.");
		}
		if (password.chars().noneMatch(Character::isUpperCase)) {
			throw new IllegalArgumentException("La contrasena debe incluir al menos una mayuscula.");
		}
		if (password.chars().noneMatch(Character::isDigit)) {
			throw new IllegalArgumentException("La contrasena debe incluir al menos un numero.");
		}
	}
}
